File: LC875.cpp
class Solution {
public:
    
    int minEatingSpeed(vector<int>& piles, int h) {

        int l = 1, ht = 1000000000;
        
        while(l < ht){
            int k = l + (ht-l)/2, t = 0;
            for(int i : piles)
                t += (i + k - 1)/k;
            if(t <= h)
                ht = k;
            else
                l = k + 1;
        }

        return l;
    }
};

File: LC875.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l = 1; ht = max(piles)

        while l < ht:
            k = l + (ht-l)//2
            if sum((p + k - 1)//k for p in piles) <= h:
                ht = k
            else:
                l = k + 1
        
        return l
        
